File: src/main/java/org/hwabeag/divesystem/events/JoinEvent.java
package org.hwabeag.divesystem.events;

import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.hwabeag.divesystem.config.ConfigManager;

public class JoinEvent implements Listener {
    FileConfiguration PlayerConfig = ConfigManager.getConfig("player");
    @EventHandler
    public void onPlayerJoin (PlayerJoinEvent event) {
        Player player = event.getPlayer();
        String name = player.getName();
        if (PlayerConfig.getString(name + ".잠수") == null) {
            PlayerConfig.addDefault(name, "");
            PlayerConfig.set(name + ".잠수", 0);
            PlayerConfig.set(name + ".잠수정보", false);
            PlayerConfig.set(name + ".페이지", 0);
            PlayerConfig.set(name + ".구매상점", "없음");
            PlayerConfig.set(name + ".구매슬롯", 0);
            PlayerConfig.set(name + ".설정상점", "없음");
            PlayerConfig.set(name + ".금액설정", false);
            PlayerConfig.set(name + ".설정슬롯", 0);
            ConfigManager.saveConfigs();
        }
    }
}

File: src/main/kotlin/org/hwabeag/divesystem/events/JoinEvent.kt
package org.hwabeag.divesystem.events

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.hwabeag.divesystem.config.ConfigManager
import org.hwabeag.divesystem.data.DatabaseManager
import org.hwabeag.divesystem.data.PlayerStateManager

class JoinEvent : Listener {
    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        val name = event.player.name
        val playerConfig = ConfigManager.getConfig("player")
        PlayerStateManager.initializeDefaults(name)
        DatabaseManager.ensurePlayer(name)
        val legacyPoints = playerConfig.getInt("$name.잠수", 0)
        if (legacyPoints > 0 && DatabaseManager.getPoints(name) == 0) {
            DatabaseManager.setPoints(name, legacyPoints)
        }
        playerConfig.set("$name.잠수", null)
        ConfigManager.saveConfigs()
    }
}
